libs/platform/ebpf_object.c

```diff
@@ -61,12 +61,17 @@ ebpf_object_tracking_terminate()
 }
 
 void
-ebpf_object_initialize(ebpf_object_t* object, ebpf_object_type_t object_type, ebpf_free_object_t free_function)
+ebpf_object_initialize(
+    ebpf_object_t* object,
+    ebpf_object_type_t object_type,
+    ebpf_free_object_t free_function,
+    ebpf_object_get_program_type_t get_program_type_function)
```

Comment:
Questions:
Does get_program_type_function apply when object_type != program?
What functions does it point to (it looks like it always points to the same one)?

Reply:
Yes.   For example it applies to a BPF_MAP_TYPE_PROG_ARRAY, and thus also to a BPF_MAP_TYPE_ARRAY_OF_MAPS which contains a prog array map, and to a BPF_MAP_TYPE_HASH_OF_MAPS which contains a prog array map, or an array of maps that contain a prog array map, etc.

Reply:
Updated comment in ebpf_object.h